soufi/finders/rhel.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2021 Cisco Systems, Inc. and its affiliates
+# All rights reserved.
+
+from typing import Iterable
+
+import soufi.finders.yum as yum_finder
+from soufi import finder
+
+DEFAULT_REPO = "https://cdn-ubi.redhat.com/content/public/ubi/dist/"
+
+
+class RHELFinder(yum_finder.YumFinder):
+    """Find Red Hat Enterprise Linux source files.
+
+    By default, uses the public UBI index at https://cdn-ubi.redhat.com/
+    """
+
+    distro = finder.Distro.rhel.value
+
+    # UBI is kind of sprawling, and they are fairly idiosyncratic with where
+    # they like to put their packages, tree-wise.  Once you find the proper
+    # subtree, they are delightfully regular with the repo structure, however.
+    # Rather than aimlessly poke around their CDN, we'll use this
+    # hand-curated list of search paths.
+    default_search_dirs = (
+        'ubi8/8/x86_64/baseos/',
+        'ubi8/8/x86_64/appstream/',
+        'ubi8/8/x86_64/codeready-builder/',
+        'ubi/server/7/7Server/x86_64/',
+        'ubi/server/7/7Server/x86_64/extras/',
+        'ubi/server/7/7Server/x86_64/devtools/1/',
+        'ubi/server/7/7Server/x86_64/optional/',
+        'ubi/server/7/7Server/x86_64/rhscl/1/',
+        'ubi/atomic/7/7Server/x86_64/',
+    )
+
+    def __init__(
+        self,
+        *args,
+        source_repos: Iterable[str] = None,
+        binary_repos: Iterable[str] = None,
+        **kwargs,
+    ):
+        if not source_repos:
+            source_repos = self._get_source_repos()
+        if not binary_repos:
+            binary_repos = self._get_binary_repos()
```

Comment:
Can this just not go in the base class? Or am I missing something?

Reply:
It could, and for a minute it did, but it made things look really awkward to my eyes, so I kept it like this for clarity.

Reply:
It looks weirder like this to my eyes... YMMV :)

Reply:
After thinking about it a bit more, it does seem rather bonkers to define an ABC that demands a pair of methods to be defined in subclasses, and then never calls those methods itself.

So I'll get over my aesthetic concerns and DRY this out, as it does seem to make the most sense.